src/components/AddressSearch.js

```diff
@@ -40,52 +40,53 @@ const defaultProps = {
-            const streetNumber = GooglePlacesUtils.getAddressComponent(addressComponents, 'street_number', 'long_name');
-            const streetName = GooglePlacesUtils.getAddressComponent(addressComponents, 'route', 'long_name');
-            let city = GooglePlacesUtils.getAddressComponent(addressComponents, 'locality', 'long_name');
-            if (!city) {
-                city = GooglePlacesUtils.getAddressComponent(addressComponents, 'sublocality', 'long_name');
-                Log.hmmm('[AddressSearch] Replacing missing locality with sublocality: ', {address: details.formatted_address, sublocality: city});
-            }
-            const state = GooglePlacesUtils.getAddressComponent(addressComponents, 'administrative_area_level_1', 'short_name');
-            const zipCode = GooglePlacesUtils.getAddressComponent(addressComponents, 'postal_code', 'long_name');
-
-            // Trigger text change events for each of the individual fields being saved on the server
-            props.onChangeText('addressStreet', `${streetNumber} ${streetName}`);
-            props.onChangeText('addressCity', city);
-            props.onChangeText('addressState', state);
-            props.onChangeText('addressZipCode', zipCode);
-        } else {
-            // Clear the values associated to the address, so our validations catch the problem
-            Log.hmmm('[AddressSearch] Search result failed validation: ', {
-                address: details.formatted_address,
-                address_components: addressComponents,
-                place_id: details.place_id,
-            });
-            props.onChangeText('addressStreet', null);
-            props.onChangeText('addressCity', null);
-            props.onChangeText('addressState', null);
-            props.onChangeText('addressZipCode', null);
+        if (!addressComponents) {
+            return;
+        }
+
+        // Gather the values from the Google details
+        const streetNumber = GooglePlacesUtils.getAddressComponent(addressComponents, 'street_number', 'long_name') || '';
+        const streetName = GooglePlacesUtils.getAddressComponent(addressComponents, 'route', 'long_name') || '';
+        const street = `${streetNumber} ${streetName}`.trim();
+        let city = GooglePlacesUtils.getAddressComponent(addressComponents, 'locality', 'long_name');
+        if (!city) {
+            city = GooglePlacesUtils.getAddressComponent(addressComponents, 'sublocality', 'long_name');
+            Log.hmmm('[AddressSearch] Replacing missing locality with sublocality: ', {address: details.formatted_address, sublocality: city});
+        }
+        const zipCode = GooglePlacesUtils.getAddressComponent(addressComponents, 'postal_code', 'long_name');
+        const state = GooglePlacesUtils.getAddressComponent(addressComponents, 'administrative_area_level_1', 'short_name');
+
+        const values = {};
+        if (street && street.length > props.value.length) {
```

Comment:
> NAB, should we trim `props.value.length` in case there is whitespace that makes it longer but less accurate?
> 
> Also, I wonder if we need to worry about a case like this where someone enters some very long address into the street input. I feel like in trying to optimize for the Apartment # case we are maybe introducing the possibility of bad street info instead of trusting the correctness of what Google Places returns.
>
> I think we can maybe wait and see if this is a problem...

I honestly don't like it either, I think the ideal case is to have the Apt number in a separate field. As @luacmartins , this also causes the autocomplete to start showing results while the user may be only trying to input the apt number and it looks weird :P. Having said that, if we wanted to do a separate input for it, I think that should be done in a separate pr/issue.